include/onnxruntime/core/providers/coreml/coreml_provider_factory.h

```diff
@@ -61,6 +61,22 @@ static const char* const kCoremlProviderOption_SpecializationStrategy = "Special
 static const char* const kCoremlProviderOption_ProfileComputePlan = "ProfileComputePlan";
 // please refer to https://developer.apple.com/documentation/coreml/mlmodelconfiguration/allowlowprecisionaccumulationongpu
 static const char* const kCoremlProviderOption_AllowLowPrecisionAccumulationOnGPU = "AllowLowPrecisionAccumulationOnGPU";
+// Specify the directory to cache any CoreML models created from the ONNX model in.
+// CoreML EP will convert onnx subgraph to CoreML model and save to disk.
+// If this path is not specified, the model will be saved to a temp directory and deleted after the session is closed.
+// otherwise, the model will be saved to the specified path and User should manage to delete the model.
+
+// we would not detect if the cached model match the onnx subgraph, so User should carefully manage the cache for a new model.
+// The cache key is generated by
+//    1. user provided key in metadata_props if exist (prefered)
+//    2. Hash of the model url the inference session was created with
+//    3. Hash of the input/output names of the model
```

Comment:
Can we add a link to the documentation as that has the example for creating the hash in the model's metadata_props?

```suggestion
// we do NOT detect if the onnx model has changed and no longer matches the cached model. 
// the user should carefully manage the cache if modifying/replacing a model.
// The cache key is generated by
//    1. User provided key in metadata_props if found (preferred)
//    2. Hash of the model url the inference session was created with
//    3. Hash of the input/output names of the model
```